File: core/src/main/java/ru/surfstudio/android/core/util/SettingsUtil.java
package ru.surfstudio.android.core.util;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import java.util.HashSet;
import java.util.Set;

/**
 * хелпер для работы с SharedPref
 */
public class SettingsUtil {
    public static final String EMPTY_STRING_SETTING = "";
    public static final int EMPTY_INT_SETTING = -1;
    public static final long EMPTY_LONG_SETTING = -1L;

    private SettingsUtil() {
        //do nothing
    }

    public static String getString(Context context, String key) {
        return getString(getDefaultSharedPreferences(context), key);
    }

    public static void putString(Context context, String key, String value) {
        putString(getDefaultSharedPreferences(context), key, value);
    }

    public static void putInt(Context context, String key, int value) {
        putInt(getDefaultSharedPreferences(context), key, value);
    }

    public static void putLong(Context context, String key, long value) {
        putLong(getDefaultSharedPreferences(context), key, value);
    }

    public static int getInt(Context context, String key) {
        return getInt(getDefaultSharedPreferences(context), key);
    }

    public static long getLong(Context context, String key) {
        return getLong(getDefaultSharedPreferences(context), key);
    }

    public static void putBoolean(Context context, String key, boolean value) {
        putBoolean(getDefaultSharedPreferences(context), key, value);
    }

    public static boolean getBoolean(Context context, String key, boolean defaultValue) {
        return getBoolean(getDefaultSharedPreferences(context), key, defaultValue);
    }

    public static String getString(SharedPreferences sp, String key) {
        return sp.getString(key, EMPTY_STRING_SETTING);
    }

    public static Set<String> getStringSet(SharedPreferences sp, String key) {
        return sp.getStringSet(key, new HashSet<String>());
    }

    public static void putStringSet(SharedPreferences sp, String key, Set<String> value) {
        SharedPreferences.Editor editor = sp.edit();
        editor.putStringSet(key, value);
        saveChanges(editor);
    }

    public static void putString(SharedPreferences sp, String key, String value) {
        SharedPreferences.Editor editor = sp.edit();
        editor.putString(key, value);
        saveChanges(editor);
    }

    public static void putInt(SharedPreferences sp, String key, int value) {
        SharedPreferences.Editor editor = sp.edit();
        editor.putInt(key, value);
        saveChanges(editor);
    }

    public static void putLong(SharedPreferences sp, String key, long value) {
        SharedPreferences.Editor editor = sp.edit();
        editor.putLong(key, value);
        saveChanges(editor);
    }

    public static int getInt(SharedPreferences sp, String key) {
        return sp.getInt(key, EMPTY_INT_SETTING);
    }

    public static long getLong(SharedPreferences sp, String key) {
        return sp.getLong(key, EMPTY_LONG_SETTING);
    }

    public static void putBoolean(SharedPreferences sp, String key, boolean value) {
        SharedPreferences.Editor editor = sp.edit();
        editor.putBoolean(key, value);
        saveChanges(editor);
    }

    public static boolean getBoolean(SharedPreferences sp, String key, boolean defaultValue) {
        return sp.getBoolean(key, defaultValue);
    }

    private static void saveChanges(SharedPreferences.Editor editor) {
        editor.apply();
    }

    private static SharedPreferences getDefaultSharedPreferences(Context context) {
        return PreferenceManager.getDefaultSharedPreferences(context);
    }

}

File: util-ktx/src/main/java/ru/surfstudio/android/utilktx/util/SettingsUtil.kt
package ru.surfstudio.android.utilktx.util

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

import java.util.HashSet

/**
 * хелпер для работы с SharedPref
 */
object SettingsUtil {
    val EMPTY_STRING_SETTING = ""
    val EMPTY_INT_SETTING = -1
    val EMPTY_LONG_SETTING = -1L

    fun getString(context: Context, key: String): String {
        return getString(getDefaultSharedPreferences(context), key)
    }

    fun putString(context: Context, key: String, value: String) {
        putString(getDefaultSharedPreferences(context), key, value)
    }

    fun putInt(context: Context, key: String, value: Int) {
        putInt(getDefaultSharedPreferences(context), key, value)
    }

    fun putLong(context: Context, key: String, value: Long) {
        putLong(getDefaultSharedPreferences(context), key, value)
    }

    fun getInt(context: Context, key: String): Int {
        return getInt(getDefaultSharedPreferences(context), key)
    }

    fun getLong(context: Context, key: String): Long {
        return getLong(getDefaultSharedPreferences(context), key)
    }

    fun putBoolean(context: Context, key: String, value: Boolean) {
        putBoolean(getDefaultSharedPreferences(context), key, value)
    }

    fun getBoolean(context: Context, key: String, defaultValue: Boolean): Boolean {
        return getBoolean(getDefaultSharedPreferences(context), key, defaultValue)
    }

    fun getString(sp: SharedPreferences, key: String): String {
        return sp.getString(key, EMPTY_STRING_SETTING)
    }

    fun getStringSet(sp: SharedPreferences, key: String): Set<String> {
        return sp.getStringSet(key, HashSet())
    }

    fun putStringSet(sp: SharedPreferences, key: String, value: Set<String>) {
        val editor = sp.edit()
        editor.putStringSet(key, value)
        saveChanges(editor)
    }

    fun putString(sp: SharedPreferences, key: String, value: String) {
        val editor = sp.edit()
        editor.putString(key, value)
        saveChanges(editor)
    }

    fun putInt(sp: SharedPreferences, key: String, value: Int) {
        val editor = sp.edit()
        editor.putInt(key, value)
        saveChanges(editor)
    }

    fun putLong(sp: SharedPreferences, key: String, value: Long) {
        val editor = sp.edit()
        editor.putLong(key, value)
        saveChanges(editor)
    }

    fun getInt(sp: SharedPreferences, key: String): Int {
        return sp.getInt(key, EMPTY_INT_SETTING)
    }

    fun getLong(sp: SharedPreferences, key: String): Long {
        return sp.getLong(key, EMPTY_LONG_SETTING)
    }

    fun putBoolean(sp: SharedPreferences, key: String, value: Boolean) {
        val editor = sp.edit()
        editor.putBoolean(key, value)
        saveChanges(editor)
    }

    fun getBoolean(sp: SharedPreferences, key: String, defaultValue: Boolean): Boolean {
        return sp.getBoolean(key, defaultValue)
    }

    private fun saveChanges(editor: SharedPreferences.Editor) {
        editor.apply()
    }

    private fun getDefaultSharedPreferences(context: Context): SharedPreferences {
        return PreferenceManager.getDefaultSharedPreferences(context)
    }

}//do nothing
